src/Build/Evaluation/Evaluator.cs

```diff
@@ -1383,6 +1418,15 @@ private void EvaluatePropertyElement(ProjectPropertyElement propertyElement)
         {
             using (_evaluationProfiler.TrackElement(propertyElement))
             {
+                if (propertyElement.Name.Equals(ReservedPropertyNames.allProjects, StringComparison.OrdinalIgnoreCase)
+                    && !Traits.Instance.EscapeHatches.EnableLegacyMSBuildAllProjects)
+                {
+                    // Ignore evaluating MSBuildAllProjects unless the escape hatch is enabled.  Also keep track of whether or not it was set
+                    // so a message can be logged indicating it was ignored.
+                    _msbuildAllProjectsWasSet = true;
```

Comment:
I'm still a bit leery of this--I'd rather see it also check for the usual pattern `$(MSBuildAllProjects);$(MSBuildThisFileFullPath)` for safety.